frontend/week4/js/homework2.js

```
@@ -0,0 +1,51 @@
+function errorMessage(errorTypeMe) {
+    document.getElementById("errorMessage").innerHTML += " Invalid " + errorTypeMe + "!! <br>";
+}
+
+
+function clearErrorMessage() {
+    document.getElementById("errorMessage").innerHTML = "";
+}
+
+function printAnswer(bmi, weightType) {
+    let name = document.getElementById("usersName").value;
+    document.getElementById("resultText").innerHTML = name + ", BMI = " + bmi + ", weight type = " + weightType;
+}
+
+function clearAnswer() {
+
+    document.getElementById("resultText").innerHTML = "";
+
+}
+
+function calculateWeightType(bmi) {
+    if (bmi < 20) {
+        return "under weight";
+    } else if (bmi > 20 && bmi < 30) {
+        return "normal weight";
+    } else {
+        return "over weight";
```

This always returns "over weight" because of the previous function didn't stop when the input numbers were wrong. What was possible was that BMI is undefined, and then this function always executes the last else.
While this should never happen, it can hide errors (like in this case). It is better to do `else if (bmi >= 30)... else {throw new error("Unexpected BMI value, "+ bmi);